### ISSUE TITLE
Error of deserialization during CheckIn

### ISSUE THREAD

**Issue by xie-qianyue** [opensourceBIM/BIMserver#548]
Hello, we have an error of deserializaion when checkin this [file](https://www.dropbox.com/s/1yfdpml99424cbs/174240904075130-12-%E6%B3%B0%E5%B7%9E%E9%87%91%E8%9E%8D%E4%B8%AD%E5%BF%83%EF%BC%88%E5%85%A8%EF%BC%89.IFC?dl=0). Here is the error log:

```
12:43:33 ERROR [LongStreamingCheckinAction]:                                                                                  (LongAction.java:189) 
org.bimserver.shared.exceptions.UserException: Error on line 463987: IfcBeam expects 8 fields, but less found (String index out of range: -1)
	at org.bimserver.database.actions.StreamingCheckinDatabaseAction.execute(StreamingCheckinDatabaseAction.java:347)
	at org.bimserver.database.actions.StreamingCheckinDatabaseAction.execute(StreamingCheckinDatabaseAction.java:79)
	at org.bimserver.database.DatabaseSession.executeAndCommitAction(DatabaseSession.java:708)
	at org.bimserver.database.DatabaseSession.executeAndCommitAction(DatabaseSession.java:685)
	at org.bimserver.longaction.LongStreamingCheckinAction.execute(LongStreamingCheckinAction.java:62)
	at org.bimserver.longaction.LongActionManager$1.run(LongActionManager.java:43)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.bimserver.plugins.deserializers.DeserializeException: Error on line 463987: IfcBeam expects 8 fields, but less found (String index out of range: -1)
	at org.bimserver.ifc.step.deserializer.IfcStepStreamingDeserializer.processRecord(IfcStepStreamingDeserializer.java:358)
	at org.bimserver.ifc.step.deserializer.IfcStepStreamingDeserializer.processLine(IfcStepStreamingDeserializer.java:246)
	at org.bimserver.ifc.step.deserializer.IfcStepStreamingDeserializer.read(IfcStepStreamingDeserializer.java:181)
	at org.bimserver.ifc.step.deserializer.IfcStepStreamingDeserializer.read(IfcStepStreamingDeserializer.java:163)
	at org.bimserver.database.actions.StreamingCheckinDatabaseAction.execute(StreamingCheckinDatabaseAction.java:195)
	... 6 common frames omitted
```

I've analyzed the problem a little bit. It occurs when deserializing the line 463988 of the Ifc file:

```
#464057= IFCBEAM('ffs_8$9BIXkm3zkvLB6aXm',#2,'AKL-1/28(13)(400*800)',$,$,#464058,#464062,'26069');
```

When calling `StringUtils.nextString(realData, lastIndex);` (in IfcStepStreamingDeserializer.java), it cannot return the correct value for the third attribute "'AKL-1/28(13)(400*800)'". The method `nextString` will fail in this pattern: `... / ... * ... ,`. 

Because:
1.when it meets the first `/`, it set the escapeMode to 5;
2.After that, it meets `*`, as the escapeMode is 5, it set it to 6;
3.Finally, when it meets `,`, as the escapteMode is 6, it won't stop, and continue to the next attribute.

Since I'm not familiar to the specification of the IFC standard, I can't fix this problem. Could you please check it whether it is a bug of the method `nextString` in the package `org.bimserver.utils`.

Thanks!

**Comment by rubendel** [comment]
Fix will be in in IfcPlugins version 0.0.28

**Comment by xie-qianyue** [comment]
The Ifc file is uploaded in success.
